Fix  #1771 Add documentation for document type extension field (#1836)
beeware/briefcase: docs/reference/configuration.rst

``[tool.briefcase.app.<app name>.<platform>]``
----------------------------------------------

> Configuration options for an application that are platform-specific. The

Reviewer: ```suggestion
Configuration options for an application that are platform specific. The
```

A hexadecimal RGB color value (e.g., ``#D81B60``) for a subtle secondary color
to be used throughout an application to call attention to key elements. This

> setting is only used if the platform allows color modification, otherwise, it

Reviewer: ```suggestion
setting is only used if the platform allows color modification, otherwise it
```

A hexadecimal RGB color value (e.g., ``#008577``) to use as the primary color
for the application. This setting is only used if the platform allows color

> modification, otherwise, it is ignored.

Reviewer: ```suggestion
modification, otherwise it is ignored.
```

``primary_color_dark``
~~~~~~~~~~~~~~~~~~~~~~

A hexadecimal RGB color value (e.g., ``#008577``) used alongside the primary
color. This setting is only used if the platform allows color modification,

> otherwise, it is ignored.

Reviewer: ```suggestion
otherwise it is ignored.
```

will be the *concatenation* of requirements from all levels, starting from least
to most specific.

> See ref:`requires <configuration-requires-key>` for examples.

Reviewer: ```suggestion
See :ref:`requires <configuration-requires-key>` for examples.
```

configuration section for each document type the application can support. This
section follows the format:

    ``[tool.briefcase.app.<app name>.document_type.<document type id>]``

or, for a platform-specific definition:

    ``[tool.briefcase.app.<app name>.<platform>.document_type.<document type id>]``

> The ``document type id`` is an identifier represented in **alphanumeric format** appended to the app_id of an application to identify documents of the same type uniquely.

Reviewer: ```suggestion
The ``document type id`` is an identifier, in alphanumeric format. It is appended to the app id of an application to identify documents of the same type.
```
